api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractsApiController.java

```diff
@@ -61,4 +67,75 @@ public ResponseEntity voteOnContract(@RequestBody @NonNull final ContractVoteReq
 
         return new ResponseEntity(HttpStatus.OK);
     }
+
+    @RequestMapping(path = "listStreamerBounties/{page}/{pageSize}", method = RequestMethod.GET)
+    public ResponseEntity listContractsForStreamer(
+            @PathVariable final int page,
+            @PathVariable final int pageSize,
+            @RequestParam("state") final Optional<String> optionalState,
+            @RequestParam("username") final Optional<String> optionalUsername) {
+        final Pageable pageable = PageRequest.of(page, pageSize);
+        Page<ContractDto> contracts = null;
+        final UserModel streamer = getUser(optionalUsername);
+        final ContractState state = getContractState(optionalState);
+        if (streamer != null && state != null) {
+            contracts = contractService.getContractsForStreamerAndState(streamer, state, pageable);
+        } else if (streamer != null) {
+            contracts = contractService.getContractsForStreamer(streamer, pageable);
+        } else if (state != null) {
+            contracts = contractService.getContractsForState(state, pageable);
+        } else {
+            contracts = contractService.getAllContracts(pageable);
+        }
+
+        return ResponseEntity.ok(contracts);
+    }
+
+    @RequestMapping(path = "listDonatorBounties/{page}/{pageSize}", method = RequestMethod.GET)
+    public ResponseEntity listContractsForDonator(
+            @Nullable final OAuth2Authentication authentication,
+            @PathVariable final int page,
+            @PathVariable final int pageSize,
+            @RequestParam("state") final Optional<String> optionalState) {
+        if (authentication == null) {
+            return new ResponseEntity(HttpStatus.FORBIDDEN);
+        }
+
+        final UserModel donator = userService.getUserFromAuthContext(authentication);
+        final Pageable pageable = PageRequest.of(page, pageSize);
+        Page<ContractDto> contracts = null;
+        final ContractState state = getContractState(optionalState);
+        if (state != null) {
+            contracts = contractService.getContractsForDonatorAndState(donator, state, pageable);
+        } else {
+            contracts = contractService.getContractsForDonator(donator, pageable);
+        }
+
+        return ResponseEntity.ok(contracts);
+    }
+
+    private ContractState getContractState(Optional<String> optionalState) {
+        if (optionalState.isPresent()) {
+            try {
+                return ContractState.valueOf(optionalState.get());
+            } catch (IllegalArgumentException e) {
+            }
```

Comment:
If we cant get rid of this method, lets log a warning to capture what invalid states someone tried to hit our endpoint with

Reply:
Got rid of it